processor/src/main/java/org/mapstruct/ap/internal/model/source/ForgedMethod.java

```diff
@@ -151,11 +93,12 @@ public ForgedMethod(String name, Type sourceType, Type returnType, MapperConfigu
      *            parameters
      * @param history a parent forged method if this is a forged method within a forged method
      * @param mappingOptions the mapping options for this method
+     * @param forgedNameBased forges a name based (matched) mapping method
      */
-    private ForgedMethod(String name, Type sourceType, Type returnType, MapperConfiguration mapperConfiguration,
+    public ForgedMethod(String name, Type sourceType, Type returnType, MapperConfiguration mapperConfiguration,
```

Comment:
Next time someone visits this class, it should be refactored to either use a builder or dedicated static creator methods with proper names... :)